lib/cln.js

```diff
@@ -0,0 +1,161 @@
+import fetch from 'node-fetch'
+import https from 'https'
+
+export const createInvoice = async ({ socket, rune, cert, label, description, msats, expiry }) => {
+  const agent = cert ? new https.Agent({ ca: Buffer.from(cert, 'base64') }) : undefined
+  const url = 'https://' + socket + '/v1/invoice'
+  const randomId = Math.floor(Math.random() * 1000)
+  const res = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Rune: rune,
+      // can be any node id, only required for CLN v23.08 and below
+      // see https://docs.corelightning.org/docs/rest#server
+      nodeId: '02cb2e2d5a6c5b17fa67b1a883e2973c82e328fb9bd08b2b156a9e23820c87a490'
+    },
+    agent,
+    body: JSON.stringify({
+      // why does CLN require a unique label?
+      label: description ? `${description} ${randomId}` : randomId,
+      description,
+      amount_msat: msats,
+      expiry
+    })
+  })
+  const inv = await res.json()
+  if (inv.error) {
+    throw new Error(inv.error.message)
+  }
+  return inv
+}
+
+// https://github.com/clams-tech/rune-decoder/blob/57c2e76d1ef9ab7336f565b99de300da1c7b67ce/src/index.ts
```

Comment:
Copied code from https://github.com/clams-tech/rune-decoder since `npm install` from Github lead to import issues (relative import `./base64` wasn't found for some reason). However, there is no license provided for this code :sweat_smile: 

Opened a PR to be fair: https://github.com/clams-tech/rune-decoder/pull/1